torch/distributed/CONTRIBUTING.md

```diff
@@ -6,71 +6,21 @@ Please go through PyTorch's top level [Contributing Guide](../../CONTRIBUTING.md
 
 In this document, we mostly focus on some of the code structure for PyTorch distributed and implementation details.
 
-## C10D and DistributedDataParallel
-
-The figure below demonstrates building blocks of the c10d and DDP package and shows how typically an application is layered on top. Most parts of the distributed package are implemented in C++ and then bound to the Python frontend (see [c10d/init.cpp](../csrc/distributed/c10d/init.cpp)).
-
-![C10D_ARCH](../../docs/source/_static/img/pt_distributed_arch.png)
-
-### Process Groups
-
-Process groups (PG) take care of communications across processes. It is up to users to decide how to place processes, e.g., on the same machine or across machines. PG exposes a set of communication APIs, e.g., send, recv, broadcast, allgather, allreduce, etc.
-
-Source Code: [ProcessGroup.cpp](../csrc/distributed/c10d/ProcessGroup.cpp) and  [ProcessGroup.hpp](../csrc/distributed/c10d/ProcessGroup.hpp)
-
-#### Process Group Backends
-
-We currently offer three backends for Process Groups: [ProcessGroupGloo.hpp](../csrc/distributed/c10d/ProcessGroupGloo.hpp), [ProcessGroupMPI.hpp](../csrc/distributed/c10d/ProcessGroupMPI.hpp) and [ProcessGroupNCCL.hpp](../csrc/distributed/c10d/ProcessGroupNCCL.hpp)
-
-#### Store
-
-Processes discover each other through a rendezvous process on a common Store (See [Store.hpp](../csrc/distributed/c10d/Store.hpp) for the interface and [FileStore.hpp](../csrc/distributed/c10d/FileStore.hpp), [TCPStore.hpp](../csrc/distributed/c10d/TCPStore.hpp) and [PrefixStore.hpp](../csrc/distributed/c10d/PrefixStore.hpp) for implementations.)
-
-### Distributed Data Parallel
```

Comment:
Shall we keep c10d and DDP items here? Or we should delete them and add a dedicated section to list our code structures that include the things we want to maintain? roughly we can give some code structure outlines about each of our components:
* c10d
* DTensor
* DDP
* FSDP2
* TP
* PP